### ISSUE TITLE
Extra space following close quotes (single or double)

### ISSUE THREAD

**Issue by mbadros** [vkbo/novelWriter#675]
Problem: Closing quotation marks or apostrophes are followed by an automatic space.

What I expect: This should not be automatic. Should be replaced by curved "smart quotes" or apostrophes but with no automatic space following the close.

What happens: The word _can't_ appears automatically as _can' t_ (with the curved apostrophe/smart quotes, but with the internal space). Likewise, I don't (which novelWriter wants to display as _don' t_) want the space to appear automatically after closing double-quotes as I am accustomed to spacing manually, for instance, at the end of a sentence. 

* Using English (US) style settings for quotation marks, etc.
* Using novelWriter 1.1 on Ubuntu 20

**Comment by vkbo** [comment]
I cannot reproduce this on my end, but perhaps an extra space is read in from the config file. My config file is rather old. I'll have a look at it.

**Comment by mbadros** [comment]
I looked at the relevant lines of the config:

fmtsinglequote = ‘, ’
fmtdoublequote = “, ”

I tried taking out the space as follows:

fmtsinglequote = ‘,’
fmtdoublequote = “,”

But it didn't change anything (and indeed, got changed back to include the space, as I just noticed when copying the lines).

Thanks!

**Comment by vkbo** [comment]
Yeah, I just checked the code, and there's a strip command on the read-in.
https://github.com/vkbo/novelWriter/blob/fe828a1c49b46034c7141d424934d70d310d2f56/nw/config.py#L947-L962

That is anyway a very strange bug. There is a test that checks the auto-replace when typing. It's one of the oldest tests I have, and it has the word "isn't" in it. The text it produces has no extra spaces and is consistently passing on Ubuntu, macOS and Windows, and has been for years.

What version of Qt are you running? (Help -> About Qt)

**Comment by mbadros** [comment]
I'm using Qt version 5.12.8 ... Linux Mint

**Comment by vkbo** [comment]
This is going to be a bit tricky to track down as I cannot reproduce it with any of my standard setups.

I don't know your programming level, but the function that does the replacing is this one: https://github.com/vkbo/novelWriter/blob/fe828a1c49b46034c7141d424934d70d310d2f56/nw/gui/doceditor.py#L1172-L1174

If I have to set up a new virtual machine, it may take me a while to look into.

**Comment by mbadros** [comment]
Thanks -- starting to look through your code. I have good familiarity with Python, but not a lot of experience understanding how a project like yours is structured. I'll also try to go back a version to see whether the bug existed there. (I likely would have noticed -- but maybe I didn't try 1.0 much before you released the new version.) The double/single quotation mark + space is consistently being treated as a single character. Delete or backspace removes both the quote mark and the space together; the arrow keys move over both characters in a single step.

**Comment by vkbo** [comment]
I've mostly worked on large simulation codes. This is my first attempt at writing a GUI app, and I've learned a lot over the 2-3 years working on this project. I'm still cleaning up bad structure choices from early on.

Reading your response from the email, and not seeing your edit, I immediately recognised that this must be a font issue. As the code reveals, it too moves one character back and inserts a single character to replace it. That's how the auto-replace works. It would be very strange if there were two characters and it still managed to replace the quote symbol.

I'm glad it was resolved!
